test/http_fakes.dart

```diff
@@ -28,14 +28,16 @@ class FakeHttpHeaders implements HttpHeaders {
   }
 
   @override
-  set ifModifiedSince(DateTime ifModifiedSince) {
-    // Format "ifModifiedSince" header with date in Greenwich Mean Time (GMT).
-    var formatted = HttpDate.format(ifModifiedSince.toUtc());
-    _set(HttpHeaders.ifModifiedSinceHeader, formatted);
+  set ifModifiedSince(DateTime? ifModifiedSince) {
+    if (ifModifiedSince != null) {
```

Comment:
This should probably throw instead of doing nothing if it sees `null`. You could use `ArgumentError.checkNotNull(ifModifiedSince);`

That would match previous behavior better (it would have had a null dereference error)

Reply:
Done.